src/FSharpVSPowerTools.Logic/NavigateToItem.fs

```diff
@@ -81,166 +156,153 @@ and
-        
-        interface INavigateToItemDisplayFactory with
-            member x.CreateItemDisplay(item) = 
-                let icon = x.iconCache.GetIconForNavigationItemKind(item.Kind)
-                NavigateToItemDisplay(item, icon, x.serviceProvider) :> _
-and
-    NavigateToItemDisplay(item: NavigateToItem, icon, serviceProvider: IServiceProvider) =
-        let extraData: NavigateToItemExtraData = unbox item.Tag
-        interface INavigateToItemDisplay with
-            member __.Name = item.Name
-            member __.Glyph = icon
-            member __.AdditionalInformation = extraData.FileName
-            member __.Description = extraData.Description
-            member __.DescriptionItems = Constants.EmptyReadOnlyCollection
-            member __.NavigateTo() = 
-                let (startRow, startCol), (endRow, endCol) = extraData.Span
-                serviceProvider.NavigateTo(extraData.FileName, startRow, startCol, endRow, endCol)
+        member __.Dispose() = processProjectsCTS.Cancel()
 
+type NavigateToItemDisplay(item: NavigateToItem, icon, serviceProvider: IServiceProvider) =
+    let extraData: NavigateToItemExtraData = unbox item.Tag
+    interface INavigateToItemDisplay with
+        member __.Name = item.Name
+        member __.Glyph = icon
+        member __.AdditionalInformation = extraData.FileName
+        member __.Description = extraData.Description
+        member __.DescriptionItems = Constants.EmptyReadOnlyCollection
+        member __.NavigateTo() = 
+            serviceProvider.NavigateTo(extraData.FileName, extraData.Span.Start.Row, 
+                extraData.Span.Start.Col, extraData.Span.End.Row, extraData.Span.End.Col)
+
+[<ExportWithMinimalVisualStudioVersion(typeof<INavigateToItemDisplayFactory>, Version = VisualStudioVersion.VS2012)>]
+type VS2012NavigateToItemDisplayFactory() =
+    [<Import(typeof<SVsServiceProvider>); DefaultValue>]
+    val mutable ServiceProvider: IServiceProvider
+    [<Import; DefaultValue>]
+    val mutable IconCache: NavigationItemIconCache
+    
+    interface INavigateToItemDisplayFactory with
+        member x.CreateItemDisplay(item) = 
+            let icon = x.IconCache.GetIconForNavigationItemKind(item.Kind)
+            NavigateToItemDisplay(item, icon, x.ServiceProvider) :> _
+
+[<Package("f152487e-9a22-4cf9-bee6-a8f7c77f828d")>]
+[<Export(typeof<INavigateToItemProviderFactory>)>]
+type NavigateToItemProviderFactory 
+    [<ImportingConstructor>]
+    (
+        openDocumentsTracker: IOpenDocumentsTracker,
+        [<Import(typeof<SVsServiceProvider>)>] serviceProvider: IServiceProvider,
+        languageService: VSLanguageService,
+        [<ImportMany>] itemDisplayFactories: seq<Lazy<INavigateToItemDisplayFactory, IMinimalVisualStudioVersionMetadata>>,
+        vsCompositionService: ICompositionService,
+        projectFactory: ProjectFactory,
+        [<Import(typeof<NavigableItemCache>)>] navigableItemCache: NavigableItemCache
```

Comment:
An unintended effect of importing this cache may be if even we disable the feature, the cache is still initialized and hooked with VS events. I think it's better to make it a lazy value and call its constructor if the feature is enabled. 

Note that we will only have a single instance of `NavigateToItemProviderFactory` by MEF.